addons/delivery/models/stock_move.py

```diff
@@ -19,8 +19,9 @@ def _cal_move_weight(self):
 
     def _get_new_picking_values(self):
         vals = super(StockMove, self)._get_new_picking_values()
-        carrier_id = self.group_id.sale_id.carrier_id.id
-        vals['carrier_id'] = any(propagate_carrier for propagate_carrier in self.rule_id) and carrier_id
+        if self.picking_type_id.code != 'incoming':
```

Comment:
What is the need of this ?

Reply:
while we create a sale order line with minus qty it will create a return transfer with a set carrier. so we only set the shipping carrier in only for "Delivery orders".